### ISSUE TITLE
"Zero length Lex item returned from '<unlexable>'" on SQL file with backslash

### ISSUE THREAD

**Issue by ifgarces** [sqlfluff/sqlfluff#4251]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I discovered that, when by accident adding a backslash character (`\`) at any point of an SQL file, an error occurs on `sqlfluff` (version 1.4.5, for Python 3.10.6, on Linux Mint 21.1) and the following warning appears:

```text
WARNING    Zero length Lex item returned from '<unlexable>'. Report this as a bug. 
```

### Expected Behaviour

It should complain about an invalid line or wrong character at line 2, i.e. where the `\` character is located at.

### Observed Behaviour

Just an error output explained below. I didn't notice the `\` characters the first seconds and I thought there could be something wrong with the installation of `sqlfluff` on my machine.

### How to reproduce

For instance, having the file `./db/foo.sql`:

```sql
CREATE INDEX ON vle_quiz_user (quiz);
CREATE INDEX ON vle_quiz_user_question (quiz_user); \
```

The following command throws error

```text
 $ sqlfluff lint --dialect postgres db/foo.sql 
WARNING    Zero length Lex item returned from '<unlexable>'. Report this as a bug. 
== [db/foo.sql] FAIL
L:   0 | P:   0 |  LXR | Fatal. Unable to lex characters: 0
All Finished 📜 🎉!
```

When removing the misplaced `\` character, the linter runs normally and exists with status `0`.

### Dialect

Postgres

### Version

1.4.5

### Configuration

Default configuration

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
